### ISSUE TITLE
Save as pdf

### ISSUE THREAD

**Issue by davidanthoff** [fredo-dedup/VegaLite.jl#10]
Is there a function to save a plot as a pdf to disc?

**Comment by fredo-dedup** [comment]
Not now, but definitely possible using the same third party lib (wkhtmltopdf) as the one that enabled the PlotPane display in Juno.

**Comment by davidanthoff** [comment]
I've started work on [NodeJS.jl](https://github.com/davidanthoff/NodeJS.jl), which would just package Node.js in a julia package and handle all deployment etc. of that. Once that is done, one could probably also just use the Vega scripts vg2png and vg2svg, and then separately worry about the pdf conversion.

About wkhtmltopdf: might be worth to also create a julia package that just installs that, right? I might take a look.

**Comment by davidanthoff** [comment]
I've got [NodeJS.jl](https://github.com/davidanthoff/NodeJS.jl) into pretty good shape now, and I've also experimented with various approaches to save things in all sorts of file formats. Here is one strategy that I have prototyped locally:
1. This package takes a dependency on NodeJS.jl and Rsvg.jl.
2. The build.jl script here calls ``npm install`` to install all of vega-lite and vega. This might be the most robust way to do things in any case, after all that seems to be the recommended way to do things from the vega folks.
3. I have a simple js file that takes a vegalite spec and compiles it into a svg format. It all works without ever touching the file format, essentially it just calls ``node`` under the hood and then uses the normal vega API.
4. We use Rsvg.jl with Cairo.jl to convert the svg file into all sorts of other file formats (pdf, png, esp etc.)

Does that sound good to you and would you? I have most of the code for this lying around here, if you think this is a good idea I would open a PR. Should I just open that PR against ``devel``? Is that branch functional right now?

**Comment by fredo-dedup** [comment]
Now possible with the v0.0.4 release.

**Comment by davidanthoff** [comment]
Great, thanks a lot for tagging this!!

**Comment by fredo-dedup** [comment]
It seems to be breaking on julia 0.4. So it is not officially tagged yet.
I raised the required version of Julia to 0.5, and I am waiting for the CI
tests results..  we'll see.

Le 15 juin 2017 9:53 PM, "David Anthoff" <notifications@github.com> a
écrit :
